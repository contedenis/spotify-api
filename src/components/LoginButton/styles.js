// @packages
import styled from 'styled-components';

const ButtonStyled = styled.button`
  background-color: #1db954;
  border-color: #1aa34a;
  border-radius: 500px;
  border-width: 0;
  border: none;
  color: #fff;
  cursor: pointer;
  font-size: 12px;
  letter-spacing: 2px;
  line-height: 1;
  max-width: fit-content;
  padding: 11px 32px 9px;
  text-decoration: none;
  text-transform: uppercase;
  white-space: normal;
`;

export {
  ButtonStyled,
};
