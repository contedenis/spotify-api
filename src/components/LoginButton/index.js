// @packages
import React from 'react';
import PropTypes from 'prop-types';

// @own
import { ButtonStyled } from './styles';

function LogginButton({
  handleLogOutClick,
  handleLoginClick,
  isLogged,
}) {
  return (
    <ButtonStyled onClick={isLogged ? handleLogOutClick : handleLoginClick}>
      {isLogged ? 'LOG OUT' : 'Login to Spotify'}
    </ButtonStyled>
  );
}

LogginButton.propTypes = {
  handleLogOutClick: PropTypes.func.isRequired,
  handleLoginClick: PropTypes.func.isRequired,
  isLogged: PropTypes.bool.isRequired,
};

export default LogginButton;
