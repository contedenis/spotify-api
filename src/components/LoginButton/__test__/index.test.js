// @packages
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';

// @own
import LoginButton from '../index';

describe('LoginButton', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <LoginButton />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('render LoginButton', () => {
    render(<LoginButton />);

    const buttonItem = screen.getByRole('button', { name: /Login to/i });
    expect(buttonItem).toBeInTheDocument();
  });

  it('render LoginButton with Logout text', () => {
    render(<LoginButton isLogged />);

    const buttonItem = screen.getByRole('button', { name: /Log out/i });
    expect(buttonItem).toBeInTheDocument();
  });

  it('onClick button', () => {
    const handleLoginClick = jest.fn();
    const handleLogOutClick = jest.fn();

    const { rerender } = render(
      <LoginButton handleLoginClick={handleLoginClick} />,
    );

    const buttonLoginItem = screen.getByRole('button', { name: /Login/i });
    userEvent.click(buttonLoginItem);
    expect(handleLoginClick).toHaveBeenCalledTimes(1);

    rerender(<LoginButton isLogged handleLogOutClick={handleLogOutClick} />);
    const buttonLogoutItem = screen.getByRole('button', { name: /Log out/i });
    userEvent.click(buttonLogoutItem);
    expect(handleLogOutClick).toHaveBeenCalledTimes(1);
  });
});
