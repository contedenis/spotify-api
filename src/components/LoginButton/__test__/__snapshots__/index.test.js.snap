// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton matches snapshot 1`] = `
<button
  class="sc-bdnylx fCXfLs"
>
  Login to Spotify
</button>
`;
